1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -39,29 +44,44 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * I will give you one clue / tip - You *will* need to handle exceptions for the final solution.
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
-
-			throw new NotImplementedException();
-		}
+            decimal result = 0.0M;
+            try
+            {
+                result = (decimal)firstValue + (decimal)secondValue + thirdValue;
+            }
+            
+            catch (Exception)
+            {
+                
+            }
```

Comment:
This is 100% valid, however with that said its not advised to "do nothing" on an exception. Though it is _basically_ the same, the following code example presents itself a bit better by explicitly handling the exception vs implicitly making it more obvious as to "what should happen upon failure". You would be amazed how many bugs happen because something was implicit vs explicit.

```
            try
		        {
                return (decimal)firstValue + (decimal)secondValue + thirdValue;
            }
            catch (Exception)
            {
                return 0m;
            }
```